src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -736,6 +742,19 @@ public static bool IsSingleton(string set)
                 return false;
         }
 
+        /// <summary>
+        /// <c>true</c> if the set contains a single character only
+        /// </summary>
+        public static bool IsDoubleton(string set)
+        {
+            if (set[FLAGS] == 0 && set[CATEGORYLENGTH] == 0 && set[SETLENGTH] == 4 && !IsSubtraction(set) &&
+                ((set[SETSTART] + 1 == set[SETSTART + 1]) &&   // if the class only has one member eg., "T", it is stored as "TU"
+                (set[SETSTART + 2] == LastChar || set[SETSTART + 2] + 1 == set[SETSTART + 3])))
+                return true;
```

Comment:
set[FLAGS] == 0 // it's not negated. `[a-z^aeiuo] ` shows negation
set[CATEGORYLENGTH] == 0 // there are no categories (like "punctuation") only raw ranges
set[SETLENGTH] == 4 // negation + count + start and end of range = 4
!IsSubtraction(set)  // not a subtraction. `[a-z-[aeiuo]]` shows subtraction
set[SETSTART] + 1 == set[SETSTART + 1] // start and end of the first range: they are stored such that the end is one past the actual end, which helps when binary searching. this tests that the range is width 1, ie represents one character.
set[SETSTART + 2] == LastChar || set[SETSTART + 2] + 1 == set[SETSTART + 3]) // same check, but if the char happens to be LastChar (\uFFFF) we cannot store the end as the character after it, because that would overflow a char

Perhaps this should be in a comment.